message-format.md

```diff
@@ -1,47 +1,106 @@
+|-----------|--------------|-------------------------------------------|----------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `serv`    | String       | `"out_bin_switch"`                        | Yes      | The name of the service the interface is part of. Service names should utilize `snake_case`.                                                                                                                                         |
+| `type`    | String       | `"evt.binary.report"`                     | Yes      | The name of the interface represented by the message. See the [Interface Format](#interface-format) section for more details.                                                                                                        |
+| `val_t`   | String       | `"bool"`                                  | Yes      | Type of the value stored in `val` property. See the list of [Value Types](#value-types) section for more details.                                                                                                                    |
+| `val`     | Any          | `3.33`                                    | Yes      | Payload of the message of which type is defined in `val_t` property.                                                                                                                                                                 |
+| `storage` | Object       | `{"sub_value": "kWh"}`                    | No       | Defines optional storage aggregation key for multi-value reports. Applies only to device reports. Parser must accept missing key or `null` value as an empty `object`.                                                               |
+| `props`   | String Map   | `{"unit":"W"}`                            | No       | An optional map of properties. Please note that all properties including numeric ones must be encoded as strings. Parser must accept missing key or `null` value as an empty `map`.                                                  |
+| `tags`    | String Array | `{"test"}`                                | No       | An optional list of tags. Parser must accept missing key or `null` value as an empty `array`.                                                                                                                                        |
+| `resp_to` | String       | `"pt:j1/mt:rsp/rt:app/rn:angry-dog/ad:1"` | No       | A topic where a response to the request will be expected. Does not apply to most of the broadcast type of events, especially asynchronous device events for which adapter will ignore this property. Parser must accept missing key. |
```

Comment:
A topic when a response to the request is expected.

Reply:
..for which adapter ignores this property.

Reply:
parser shall accept the missing key.

Reply:
Topic is a place where we expect the response, not time when we expect the response.

Must and shall are interchangeable in documentation in accordance with https://www.ietf.org/rfc/rfc2119.txt

The natural language differences between have to / must / shall do not apply to technical documentation, AFAIK.